src/services/sessionExec.js

```diff
@@ -366,6 +366,41 @@ const joinSession = async ({ shortname }) => {
   }
 }
 
+/**
+ * Resets a QuestionBlock in a running session
+ * @param {*} param0
+ */
+const resetQuestionBlock = async ({ id, instanceIds }) => {
+  const session = await getRunningSession(id)
+  instanceIds.forEach(async instanceId => {
+    const type = await responseCache.hget(`instance:${instanceId}:info`, 'type')
+    const questionInstance = await QuestionInstanceModel.findOne({ _id: instanceId })
+    const question = await QuestionModel.findOne({ _id: questionInstance.question })
+    if (type === QUESTION_TYPES.SC || type === QUESTION_TYPES.MC) {
+      let maxChoices = 0
+      if (type === QUESTION_TYPES.SC) {
+        maxChoices = question.versions[question.__v].options.SC.choices.length
```

Comment:
I think MongoDB __v does not necessarily correlate with the Klicker question version. Would be better to use a version identifier that is stored in Info, or to check the length of the redis hash directly (hgetall minus one for the participants key).

Reply:
Also, if the length were calculate directly from redis, would not need to treat sc and mc questions differently.

Reply:
This could also save us the database lookup for each instance, as the type is already in redis info.

Reply:
Updated deleting loops and MongoDB request removed